### PR TITLE
[Feature] OCTable events

### PR DESCRIPTION
**Pull request by HoaiLee**
## Summary by CodeRabbit

- **New Features**
  - Introduced tracking of changed fields in data tables and forms.
  - Added new events for filter and search query changes.
  - Implemented a feature to display changed fields to the user.
  - Added conditional rendering of upload options based on a new property.

- **Enhancements**
  - Improved event handling for data table filters and search queries.
  - Enhanced form update mechanism to include field names in events.

- **Bug Fixes**
  - Fixed an issue where removed filters were not properly emitting events.

- **Refactor**
  - Streamlined filter application and removal functions for clarity and performance.

- **Documentation**
  - Updated storybook previews to reflect new event handling and component properties.

Please note that these release notes are a high-level overview and do not disclose internal code changes or file names.

### CODE REVIEW

File: packages/@orchidui-vue/src/Builder/DataTable/OcDataTable.stories.js [hunk 1]
```diff
@@ -34,21 +36,26 @@ export const Default = {
...
+      return { args, updateFilterData, onClickRow, changedFields };
     },
     template: `
           <Theme class="p-8">
             <div>
               <ul>
                 <li v-for="(item, key) in args.filter">{{ key }} : {{ item }}</li>
               </ul>
+              <div>
+                Filter fields changed: {{ changedFields }}
```
Comment (kewcoder) [inline packages/@orchidui-vue/src/Builder/DataTable/OcDataTable.stories.js:57]: @HoaiLee filter field is changed, what is this for? why not use the filter model v-bind:filter and @update:filter=, is there a problem in the dashboard.

File: packages/@orchidui-vue/src/Builder/Form/OcFormBuilder.vue [hunk 2]
```diff
@@ -47,8 +47,8 @@ const props = defineProps({
 
 const emit = defineEmits(["onUpdate"]);
 
-const onUpdate = (form, value) => {
-  emit("onUpdate", form, value);
+const onUpdate = (form, value, fieldName) => {
```
Comment (kewcoder) [inline packages/@orchidui-vue/src/Builder/Form/OcFormBuilder.vue:50]: we can access fieldName using form.name | we have string and object type for name.

**Review by kewcoder — CHANGES_REQUESTED** [review]
@HoaiLee onUpdate send all form object { name: '', props: {} } and new value

File: packages/@orchidui-vue/src/Builder/Form/OcFormBuilder.vue (hunk 2 above)
Comment (HoaiLee) [reply packages/@orchidui-vue/src/Builder/Form/OcFormBuilder.vue:50]: Thank you @kewcoder you're right/ updated 🙏

File: packages/@orchidui-vue/src/Builder/DataTable/OcDataTable.stories.js (hunk 1 above)
Comment (kewcoder) [reply packages/@orchidui-vue/src/Builder/DataTable/OcDataTable.stories.js:57]: oh isee, for tracker

**Review by kewcoder — APPROVED** [review]

**Pull request merged**